frontend/src/components/DocumentList.js

```diff
@@ -1,40 +1,34 @@
-import React, { Component } from 'react'
-import { Table } from 'reactstrap'
-import DocumentListItem from './DocumentListItem'
-
-class DocumentList extends Component {
-  constructor(props) {
-    super(props)
-    this.state = {
-      documents: this.props.documents,
-      status: this.props.status
-    }
-  }
-
-  render() {
-    return (
-      <Table>
-        <caption>{this.state.status}</caption>
-        <tbody>
-          <tr>
-            <th>DOC NAME</th>
-            <th>FILE</th>
-            <th />
-          </tr>
-          {this.state.documents &&
-            this.state.documents.map((document, index) => (
-              <DocumentListItem
-                key={index}
-                document={document}
-                docClass={document.docClass}
-                fileName={document.fileName}
-                fileId={document.fileId}
-              />
-            ))}
-        </tbody>
-      </Table>
-    )
-  }
-}
-
-export default DocumentList
+import React, { Component } from 'react'
+import { Table } from 'reactstrap'
+import DocumentListItem from './DocumentListItem'
+
+class DocumentList extends Component {
+  constructor(props) {
+    super(props)
+  }
+
+  render() {
+    return (
+      <Table>
+        <caption>{this.props.status}</caption>
+        <tbody>
+          <tr>
+            <th>DOC NAME</th>
+            <th>FILE</th>
+            <th />
+          </tr>
+          {this.props.documents.map(document => (
```

Comment:
So it seems like there's a small issue surrounding this code segment here -- there's a case when the props are null, which means that the map function will cause an error and that's what's happening with the test cases. I believe there should be a few ways around this: you could either have a default state that gets loaded when there are no props and perform conditional rendering here or look into functions like getDerivedStateFromProps